json/parse.go

```diff
@@ -42,6 +55,19 @@ func skipSpacesN(b []byte) ([]byte, int) {
 	return nil, 0
 }
 
+func trailingSpaces(b []byte) int {
```

Comment:
Could we have the same optimization we use in `skipSpaces` to have the first check inlined for the case where there are no trailing spaces?

Reply:
Just ;)

```
json/parse.go:58:6: can inline trimTrailingSpaces with cost 78 as: func([]byte) []byte { if len(b) > 0 && b[len(b) - 1] <= 32 { b = trimTrailingSpacesN(b) }; return b }
```